python/pdo/contract/replication.py

```diff
@@ -46,68 +43,78 @@
 __stop_service__ = False
 
 # -----------------------------------------------------------------
-def start_replication_service():
-
-    # start the manager
-    __replication_manager_executor__.submit(__replication_manager__)
+def start_replication_service(config = {}) :
+    # make sure the service is only started once
+    global replication_task_queue
```

Comment:
Isn't this (use of locks) an overkill?, why will the service be started more than once? The function is called by response.py, and called only once. Response preserves the fact that it was called. And response runs in a single thread. (In other words, no need for this lock).

Reply:
i KNOW there were serious problems with the old logic because it was missing a bunch of locks that were necessary. for the most part, we get a free pass because of the GIL but the old code was making too many assumptions about being single threaded. i'll take the locks around global variables that are part of threading.
in this case, a multi-threaded client could cause problems.